File: platform/java/agent-api/src/main/java/com/ghatana/agent/api/HealthStatus.java
/*
 * Copyright (c) 2026 Ghatana Inc.
 * All rights reserved.
 */
package com.ghatana.agent.api;

/**
 * Operational status of an agent during its lifecycle.
 *
 * @doc.type enum
 * @doc.purpose Agent lifecycle status
 * @doc.layer core
 * @doc.pattern ValueObject
 */
public enum HealthStatus {

    /** Fully operational and accepting input. */
    HEALTHY,

    /** Operational but experiencing degraded performance. */
    DEGRADED,

    /** Not operational. All processing requests will fail. */
    UNHEALTHY,

    /** Initializing — not yet accepting input. */
    STARTING,

    /** Shutting down — will stop accepting new input. */
    STOPPING,

    /** Status unknown (e.g., health check timed out). */
    UNKNOWN
}

File: platform/typescript/kernel-product-contracts/src/health/HealthStatus.ts
/**
 * HealthStatus - health status for health snapshots.
 *
 * @doc.type type
 * @doc.purpose Health status enumeration for health snapshots
 * @doc.layer kernel-product-contracts
 * @doc.pattern ValueObject
 */

import { z } from "zod";

/**
 * Health status for health snapshots.
 */
export type HealthStatus =
  | "healthy"
  | "degraded"
  | "blocked"
  | "failed"
  | "skipped"
  | "unknown"
  | "requires-approval"
  | "requires-verification"
  | "obsolete"
  | "quarantined";

export const HealthStatusSchema = z.enum([
  "healthy",
  "degraded",
  "blocked",
  "failed",
  "skipped",
  "unknown",
  "requires-approval",
  "requires-verification",
  "obsolete",
  "quarantined",
]);

export function validateHealthStatus(value: unknown): value is HealthStatus {
  return HealthStatusSchema.safeParse(value).success;
}
